server/src/main/java/org/elasticsearch/index/search/QueryStringQueryParser.java

```diff
@@ -417,11 +417,11 @@ protected Query getRangeQuery(String field, String part1, String part2,
     private Query getRangeQuerySingle(String field, String part1, String part2,
                                       boolean startInclusive, boolean endInclusive, QueryShardContext context) {
         MappedFieldType currentFieldType = context.fieldMapper(field);
-        if (currentFieldType == null) {
+        if (currentFieldType == null || currentFieldType.getTextSearchInfo() == TextSearchInfo.NONE) {
```

Comment:
This change makes range query and fuzzy query behave the same way as prefix query